Sources/TuistGenerator/Generator/WorkspaceGenerator.swift

```diff
@@ -70,7 +70,7 @@ final class WorkspaceGenerator: WorkspaceGenerating {
         logger.notice("Generating workspace \(workspaceName)", metadata: .section)
 
         /// Projects
-        let projects = try graph.projects.map { project in
+        let projects = try graph.projects.concurrentMap { project in
```

Comment:
@kwridan what do you think about changing the implementation a bit so that we can inject the threading behavior when initializing the class? The downside of defining it in an extension, is that unit tests become integration tests because we are testing the integration of our logic with the threading system. I've seen that causing flakiness in some codebases so I'd avoid falling into the same trap.

I'd recommend following the idea [on this article](https://www.cocoawithlove.com/blog/specifying-execution-contexts.html). We can define execution contexts that we can pass to the map:

```swift
try graph.projects.map(mapContext: self.context) { project in
  // Logic
}
```

Reply:
Sure will try that out 👍